parking/backend/engine/alloc_engine.py

```diff
@@ -0,0 +1,54 @@
+import logging
+from typing import Optional
+from geopy.distance import distance
+from parking.backend.db.dbaccess import DbAccess
+from parking.shared.rest_models import ParkingLot
+from parking.shared.location import Location
+from parking.shared.ws_models import ParkingRequestMessage
+
+logger = logging.getLogger('backend')
+
+
+class AllocationEngine():
+    def __init__(self, dba: DbAccess, user_sessions: 'UserSessions') -> None:
+        self.dba = dba
+        self.user_sessions = user_sessions
+
+    async def handle_allocation_request(self, user_id: str, request: ParkingRequestMessage) -> Optional[ParkingLot]:
+        rejections = self.user_sessions.get_user(user_id).rejections
+        lots = await self.dba.get_available_parking_lots(request.location, 100, rejections)
```

Comment:
Try to get 'distance' from ParkingRequestMessage.prefereces['distance'], if not then use a default value larger than 100 meters.

Reply:
I'll check for this, it might be nice to create a model for preferences in the future.

I think that ```handle_allocation_message``` should probably raise an ```InvalidDistanceException``` that the caller will handle - thoughts?

Reply:
Also isn't this call inconsistent with the signature given earlier:

    async def get_available_parking_lots(self, lat: float, long: float,
                                         dist_meters: int, exclusions: List[int]) -> List[Record]:

Reply:
Yeah, that's fixed in the newest version - we went with passing a ```Location``` object instead.

Reply:
I see, is the existence in this commit an artefact of your rebasing shenanigans by any chance